src/binder/assemblybinder.cpp

```diff
@@ -1158,13 +1166,48 @@ namespace BINDER_SPACE
                                                                 pBindResult));
             }
         }
-        else
+        else 
         {
-            // Is assembly on TPA list?
+            ReleaseHolder<Assembly> pTPAAssembly;
             SString &simpleName = pRequestedAssemblyName->GetSimpleName();
+
+            // Is assembly in the bundle?
+            // Single-file bundle contents take precedence over TPA.
+            // Bundled assemblies are treated similar to TPAs.
+            if (g_BinderVariables->fIsBundle)
```

Comment:
I'm wondering why do we even need this. The TPA should contain the bundle assemblies anyway - and the code below to handle the TPA will end up calling the GetAssembly with the TPA file path. Currently the BundleInfo will actually deal with full file paths as well (it matches the prefix), but even if it would not, it would make sense to have the TPA modified to contain the "bundle paths".

Since you implemented the bundle hookup in the PEImage all the callers to the PEImage can be pretty much oblivious to whether the file comes from bundle or not.

Reply:
The bundle meta-data is not reflected in the TPA. For self-contained single-file apps, typically the TPA is just the `app.dll`. 

I don't think we should add the bundled contents to the TPA -- it is unnecessary to search for the bundle contents in two places. In a full implementation, it is better to retire the TPA and use a common host-callback instead.

Reply:
Thanks I think I finally understand the change 😄 
Can you please extend the comment to mention:
* TPA is intentionally incomplete for bundles
* Bundled assemblies are not listed in TPA - and so we first probe the bundle
* The code will call `GetAssembly` with just a file name, no path